webapp/web-components/cht-form/src/app.component.ts

```diff
@@ -0,0 +1,183 @@
+import { Component, EventEmitter, Input, Output } from '@angular/core';
+import { EnketoFormContext, EnketoService } from '@mm-services/enketo.service';
+import * as medicXpathExtensions from '../../../src/js/enketo/medic-xpath-extensions';
+import moment from 'moment';
+import { toBik_text } from 'bikram-sambat';
+import { TranslateService } from '@mm-services/translate.service';
+
+@Component({
+  selector: 'cht-form',
+  templateUrl: './app.component.html',
+})
+export class AppComponent {
+  private readonly DEFAULT_FORM_ID = 'cht-form-id';
+  private readonly DEFAULT_USER = { contact_id: 'default_user', language: 'en' } as const;
+  private readonly DEFAULT_STATUS = {
+    saving: false as boolean,
+    error: null as string | null,
+  } as const;
+
+  private _formId = this.DEFAULT_FORM_ID;
+  private _formXml?: string;
+  private _formModel?: string;
+  private _formHtml?: string;
+  private _contactSummary?: Record<string, any>;
+  private _content: Record<string, any> | null = null;
+  private _user: Record<string, any> = this.DEFAULT_USER;
```

Comment:
```suggestion
  private _user: typeof this.DEFAULT_USER & Record<string, any> = this.DEFAULT_USER;
```
for a bit of autocomplete